File: Java/Basics/file12.java

// factorial of a number .......

import java.util.*;

public class file12 {

    static int calc_factorial(int num) {
        if(num == 0 || num == 1) {
            return 1;
        } else {
            return num * calc_factorial(num - 1);
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int num = sc.nextInt();
        int fact = 0;

        fact = calc_factorial(num);

        System.out.println(fact);
    }
}

File: C/Basics/file12.c
//Factorial of a number ....

int factorial_calc(int num) {
    if(num == 0 || num == 1) {
        return 1;
    } else {
        return num * factorial_calc(num - 1);
    }
}

#include<stdio.h>

int main()
{
    int num, fact ;
    
    scanf("%d", &num);

    fact = factorial_calc(num);

    printf("%d",fact);
}